### PR TITLE
Add AWS::Logs::* Resource Providers to CloudFormation

### PR DESCRIPTION
**Pull request by Morijarti**
## Changes
Added:
-  "AWS::Logs::LogGroup"
-  "AWS::Logs::LogStream"
-  "AWS::Logs::SubscriptionFilter"

### CODE REVIEW

File: localstack/services/logs/resource_providers/aws_logs_loggroup.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+        IAM permissions required:
+          - logs:DescribeLogGroups
+          - logs:CreateLogGroup
+          - logs:PutRetentionPolicy
+          - logs:TagLogGroup
+          - logs:GetDataProtectionPolicy
+          - logs:PutDataProtectionPolicy
+          - logs:CreateLogDelivery
+          - s3:REST.PUT.OBJECT
```
Comment (simonrw) [inline localstack/services/logs/resource_providers/aws_logs_loggroup.py:61]: This is an interesting permission!

File: localstack/services/cloudformation/engine/quirks.py [hunk 2]
```diff
@@ -13,6 +13,8 @@
 PHYSICAL_RESOURCE_ID_SPECIAL_CASES = {
     # Example
     # "AWS::ApiGateway::Resource": "/properties/ResourceId",
+    "AWS::Logs::LogStream": "/properties/LogStreamName",
```
Comment (simonrw) [inline localstack/services/cloudformation/engine/quirks.py:16]: Interesting deviation from the spec here cc @dominikschubert

**Review by simonrw — APPROVED** [review]
Looks good, interesting examples of the non-spec-compliance of CFn.. :joy: and it's [open source](https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-logs/tree/master/aws-logs-subscriptionfilter) as well!

**Review by pinzon — APPROVED** [review]
LGTM

**Pull request merged**